pkg/cluster/config/types.go

```diff
@@ -17,63 +17,61 @@ limitations under the License.
 package config
 
 import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/kind/pkg/kustomize"
 )
 
-// NOTE: if you change these types you likely need to update
-// Validate() and DeepCopy() at minimum
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // Config contains cluster creation config
 // This is the current internal config type used by cluster
 // Other API versions can be converted to this struct with Convert()
 type Config struct {
+	metav1.TypeMeta
+
 	// Image is the node image to use when running the cluster
 	// TODO(bentheelder): split this into image and tag?
-	Image string `json:"image,omitempty"`
+	Image string
 	// KubeadmConfigPatches are applied to the generated kubeadm config as
 	// strategic merge patches to `kustomize build` internally
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/strategic-merge-patch.md
 	// This should be an inline yaml blob-string
-	KubeadmConfigPatches []string `json:"kubeadmConfigPatches,omitempty"`
+	KubeadmConfigPatches []string
 	// KubeadmConfigPatchesJSON6902 are applied to the generated kubeadm config
 	// as patchesJson6902 to `kustomize build`
-	KubeadmConfigPatchesJSON6902 []kustomize.PatchJSON6902 `json:"kubeadmConfigPatchesJson6902,omitempty"`
+	KubeadmConfigPatchesJSON6902 []kustomize.PatchJSON6902
```

Comment:
I'm not the most familiar with the k8s API machinery internals, but what's the intention with removing omitempty?

Reply:
This is on the internal API type, which never gets encoded/decoded from external (JSON) representations, so it's okay here 😄

Reply:
it doesn't _currently_ but I don't see any external type? at some point I was going to save the config to nodes and include it in the log dumps for debugging.

Reply:
:man_facepalming: I see those now. this is what I get for trying to rush through a first pass over my PR backlog